x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/support/Hasher.java

```diff
@@ -136,6 +144,182 @@ public boolean verify(SecureString text, char[] hash) {
         }
     },
 
+    BCRYPT10() {
+        @Override
+        public char[] hash(SecureString text) {
+            String salt = BCrypt.gensalt(10);
+            return BCrypt.hashpw(text, salt).toCharArray();
+        }
+
+        @Override
+        public boolean verify(SecureString text, char[] hash) {
+            String hashStr = new String(hash);
+            if (!hashStr.startsWith(BCRYPT_PREFIX)) {
+                return false;
+            }
+            return BCrypt.checkpw(text, hashStr);
+        }
+    },
+
+    BCRYPT11() {
+        @Override
+        public char[] hash(SecureString text) {
+            String salt = BCrypt.gensalt(11);
+            return BCrypt.hashpw(text, salt).toCharArray();
+        }
+
+        @Override
+        public boolean verify(SecureString text, char[] hash) {
+            String hashStr = new String(hash);
+            if (!hashStr.startsWith(BCRYPT_PREFIX)) {
+                return false;
+            }
+            return BCrypt.checkpw(text, hashStr);
+        }
```

Comment:
Maybe factor these `#verify` as private static functions, similar to `verifyPbkdf2Hash`.